src/multio/action/scale/Scale.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "multio/action/ChainedAction.h"
+#include "multio/config/ComponentConfiguration.h"  // Ensure this include is present
+
+#include "multio/action/scale/Mapping.h"
+#include "multio/action/scale/Scaling.h"
+
+namespace multio::action {
+
+class Scale final : public ChainedAction {
+public:
+    explicit Scale(const ComponentConfiguration& compConf);  // Constructor declaration
+    void executeImpl(message::Message) override;
+
+private:
+    ScaleScaling scaling_;
+    ScaleMapping mapping_;
+    std::set<std::string> paramsToScale_;
+
+    template <typename Precision>
+    message::Message ScaleMessage(message::Message&& msg) const;
```

Comment:
Why not just pass by lvalue reference and return `void`. I think it is simpler and safer in this case.

Reply:
sorry overlooked the first time I checked the comments. I agree can be reference. I'll try and test that